apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -1233,9 +1238,13 @@ def create_port_precommit(self, context):
         port = context.current
         network = context.network.current
         if (network.get('router:external') and
-                not self._is_nat_enabled_on_ext_net(network) and
                 port.get('device_owner').startswith('compute:')):
-            raise VMsDisallowedOnExtNetworkIfNatDisabled(net=network['name'])
+            if not self._is_nat_enabled_on_ext_net(network):
+                raise VMsDisallowedOnExtNetworkIfNatDisabled(
+                    net=network['name'])
+            elif self._is_edge_nat(
+                self.apic_manager.ext_net_dict.get(network['name'])):
```

Comment:
when the elif condition is getting executed, that means self._is_nat_enabled_on_ext_net(network) already returns true. That again means self.apic_manager.ext_net_dict.get(network['name']) will return something valid so we should be good here.